packages/thirdweb/src/react/core/hooks/contract/useSendTransaction.ts

```diff
@@ -127,7 +125,7 @@ export function useSendTransactionCore(
 export async function getTotalTxCostForBuy(tx: PreparedTransaction) {
   // Must pass 0 otherwise it will throw on some chains
   const gasCost = await estimateGasCost({
-    transaction: { ...tx, value: 0n },
+    transaction: tx, // NEED A PROPER FIX HERE !!!
```

Comment:
what was the issue here?

Reply:
The `value: 0n` override was added because `estimateGasCost` was throwing errors on Optimism and Base without it.